backend/src/main/scala/bloop/CompilerCache.scala

```diff
@@ -50,19 +50,21 @@ final class CompilerCache(
     logger: Logger
 ) {
 
-  private val scalaCompilerCache = new ConcurrentHashMap[ScalaInstance, ScalaCompiler]()
-
+  private val scalaInstanceCache = new ConcurrentHashMap[ScalaInstance, ScalaInstance]()
```

Comment:
Cache the `ScalaInstance` instead of the `ScalaCompiler` because two projects can have the same `ScalaInstance` but different `ClasspathOptions`.